autoload/neomake/makers/ft/rust.vim

```diff
@@ -34,7 +34,7 @@ function! s:get_cargo_workspace_root() abort
         if !empty(cd_back_cmd)
             exe cd_back_cmd
         endif
-        if v:shell_error
+        if v:shell_error || output ==# ''
```

Comment:
Why is the output check needed?
Is there something on stderr then in this case?  (check with `2>&1` at the end of them `cmd`)

Reply:
This test (https://github.com/neomake/neomake/pull/2143/files#diff-563dcda53b905303c16ad35d06d7b658R12) generated empty output which made the json decode function fail. But the test is probably invalid. I don't really understand how the tests work.

Reply:
Yeah.. that tests creates an exe, but the output is not what you normally get.
I think using `NeomakeTestsGetMakerWithOutput` instead would be better.
You can run it as project maker using `CallNeomake 0, maker`.

The tests are messy/grown in several places, so feel free to ask, add notes etc.